### PR TITLE
Plumb GET query params down to the do_get function, while cleaning things up

### PR DESCRIPTION
**Pull request by henn**
Also:
- Remove do_request() (it's not needed and wasn't used outside of these 4 functions)
- Don't return values (lose/lose scenario: nothing was generating return values
  and nothing was checking)

### CODE REVIEW

**Comment by henn** [comment]
Also, this will still need some manual sanity checking.

**Comment by zenhack** [comment]
I actually think that expresson in do_request is big enough to be worth wrapping. My take would be actually to get rid of `do_{get,put,post,...}`, and just change the calls to those functions to e.g. `do_request('GET', ...)`

**Comment by henn** [comment]
Are you thinking of the comment or the check_status_code() wrapper?
Without the return's (which shouldn't be there anyhow), I'm not sure this function helps much now. Also, separating them means we don't need to have a function that takes both a `params` and a `data` argument (which should be mutually exclusive and thus enforced if we did still do that).

Also keep in mind these do_functions:
a) are internal (only used within cli.py), and
b) will likely be replaced by the client library soon.

**Comment by zenhack** [comment]
I meant the check_status_code wrapper. Don't feel terribly strongly though.

**Comment by shuwens** [comment]
I am happy with it. +1

**Comment by shuwens** [comment]
I don't have a strong opinion regarding using unified `do_request` or `do_get`s separately. Since it make sense both ways and we might have cli library/else happening, I would suggest keep it like that and discuss the change later.

File: haas/cli.py [hunk 1]
```diff
@@ -231,34 +231,29 @@ def object_url(*args):
...
+#    Arguments:
+#
+#        `url` - The url to make the request to
+#        `data` - the body of the request (for PUT, POST and DELETE)
+#        `params` - query parameters (for GET)
 
-def do_put(url, data={}):
-    return do_request('PUT', url, data=json.dumps(data))
+def do_put(url, data=None):
+    check_status_code(http_client.request('PUT', url, data=data))
```
Comment (henn) [inline haas/cli.py]: Do we need the json.dumps() here?

@shwsun  - could you look into whether the requests and keystone -middleware libraries will automatically convert to json?

**Pull request merged**